Angular/src/app/app.component.ts

```diff
@@ -0,0 +1,111 @@
+import {NgModule, Component, enableProdMode} from '@angular/core';
+import {BrowserModule} from '@angular/platform-browser';
+import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
+import {CustomSchedulerComponent} from "./components/custom-scheduler/custom-scheduler.component";
+import {FilterFormComponent} from "./components/filter-form/filter-form.component";
+import {DxSchedulerModule, DxFormModule} from 'devextreme-angular';
+import {OptionChangedEvent} from "devextreme/ui/form";
+import {startViewDate, endViewDate} from './config';
+import {Appointment, Service, data} from "./data";
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.css'],
+  providers: [Service],
+})
+
+export class AppComponent {
+  dataSource: Appointment[];
+  currentDate: Date = new Date(2022, 9, 1);
+  useDisable: boolean = false;
+  startDate: Date = startViewDate;
+  endDate: Date = endViewDate;
+  filterValue: string = '';
+
+  constructor(service: Service) {
+    this.dataSource = service.getAppointments();
+  }
+
+  onFilterValueChange({event}: any) {
```

Comment:
Let's remove these single methods for each filter value:
 - onFilterValueChange
 - onCheckboxChange
 - onStartDateChange
 - onEndDateChange

And write one universal method for all of them:

```ts
  onFilterValuesChange(newFilterValues: FilterValues): void {
    this.filterValues = newFilterValues;
    // see the next comment about private methods
    this.dataSource = this.createFilteredAppointments(this.filterValues);
  }
```